### PR TITLE
Feat/person components/update

### LINKED ISSUE #978: update interface for person components
- [x] avatar
- [x] card
- [x] list-item
- [x] search
- [x] provider

avatar example, map all props correctly

```ts
export type PersonAvatarElementProps = PersonInfoControllerHostAttributes & {
  size?: AvatarSize;
  showFloatingOn?: PersonAvatarShowCardOnType;
  clickable?: boolean;
  disabled?: boolean;
};
```

### PR DESCRIPTION
**Pull request by AndrejNikolicEq**
# Component Development
### Update of person react components, and storybook

close #978 

### Type

- [x] Feature
- [x] Fix
- [ ] Hotfix (should release ASAP)
- [ ] Maintenance (deps only)

### Description of assignment
- Updated react components to newly created web components
- Added new component called "PersonSelect"

### CODE REVIEW

File: packages/person/package.json [hunk 1]
```diff
@@ -37,7 +40,8 @@
     "url": "https://github.com/equinor/fusion-react-components/issues"
   },
   "dependencies": {
+    "@equinor/fusion-wc-person": "^1.2.1",
     "@equinor/fusion-react-button": "^0.8.3",
-    "@equinor/fusion-wc-person": "^0.6.1"
+    "@equinor/fusion-react-utils": "^2.0.5"
   }
-}
+}
```
Comment (asbjornhaland) [inline packages/person/package.json:47]: Please insert newline at end of file. Should be a setting in your IDE.

File: storybook/src/stories/person/card-example.stories.tsx [hunk 2]
```diff
@@ -86,33 +87,34 @@ export const Size: Story<{ sizes: Array<CardProps['size']> }> = (props: { sizes:
...
-                      name: 'Wubba Lubba Dub Dub',
-                      project: {
-                          id: '1234-1234',
-                          name: 'The Citadel',
-                      },
-                  },
-              ],
-          }} />
+  <PersonProvider personResolver={createResolve}>
+    <PersonCard {...props} azureId='9876-5432-1098' dataSource={{
```
Comment (asbjornhaland) [inline storybook/src/stories/person/card-example.stories.tsx:91]: azure can be omitted when dataSource is defined.

File: storybook/src/stories/person/card-component.stories.mdx [hunk 3]
```diff
@@ -66,22 +66,65 @@ You can create your own [`<PersonProvider>`](?path=/docs/data-person-provider--p
...
+`personResolver`          | `PersonResolver`                | Person resolver for a provider. 
 
 ### Methods for PersonCard
 
 Name                                    | Return Type                     | Description
 --------------------------------------  | -----------------------------   | -----------------------------
-`getCardDetailsByAzureId(azureId)`      | `CardData**`                    | Returns `CardData` for unique AuzureId.
-`getCardDetailsByUpn(upn)`              | `CardData**`                    | Returns `CardData` for unique email address.
+`getDetails()`                          | `PersonDetails***`              | Returns `PersonDetails` of person.
+`getPhoto()`                            | `PersonPicture****`             | Returns `PersonPicture` of person.
```
Comment (asbjornhaland) [inline storybook/src/stories/person/card-component.stories.mdx:76]: Copy paste? Are these the correct methods?

File: storybook/src/stories/person/list-item-component.stories.mdx [hunk 4]
```diff
@@ -63,19 +63,36 @@ You can create your own [`<PersonProvider>`](?path=/docs/data-person-provider--p
...
+`personResolver`          | `PersonResolver`                | Person resolver for a provider. 
 
 ### Methods for PersonListItem
 
 | Name                                   | Return Type                     | Description
 | -------------------------------------- | -----------------------------   | -----------------------------
-| `getListItemDetailsByAzureId(azureId)` | `ListItemData*`                | Returns `ListItemData` for unique AuzureId.
-| `getListItemDetailsByUpn(upn)`         | `ListItemData*`                | Returns `ListItemData` for unique AuzureId.
+`getInfo()`                              | `PersonInfo`                    | Returns `PersonInfo` of person.
+`getPhoto()`                             | `PersonPicture**`                | Returns `PersonPicture` of person.
```
Comment (asbjornhaland) [inline storybook/src/stories/person/list-item-component.stories.mdx:73]: Copy paste? Are these the correct methods?

File: storybook/src/stories/person/resolve-mock/PersonResolve.tsx [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+type PersonPicture = PersonDetails & { pictureSrc: string };
+
+export const createResolve: PersonResolver = {
+    getDetails: (args) => fakeResponseTime().then(() => getUsers(args) as PersonDetails),
+    getInfo: (args) => fakeResponseTime().then(() => getUsers(args) as PersonInfo),
+    getPhoto: (args) =>
+        Promise.resolve((getUsers(args) as PersonPicture).pictureSrc),
+    search: (args) => fakeResponseTime(1000).then(() => getUsers(args) as PersonSearchResult),
+};
```
Comment (asbjornhaland) [inline storybook/src/stories/person/resolve-mock/PersonResolve.tsx:177]: new line.

File: storybook/src/stories/person/select-example.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    </div>
+  </PersonProvider>
+);
+Component.args = {
+  placeholder: 'Start to type to search...',
+  initialText: 'The initial text result',
+  variant: "page-outlined",
+  dropdownHeight: '300px',
+  leadingIcon: 'search',
+};
```
Comment (asbjornhaland) [inline storybook/src/stories/person/select-example.stories.tsx:87]: new line

**Review by asbjornhaland — CHANGES_REQUESTED** [review]

File: storybook/src/stories/person/card-example.stories.tsx (hunk 2 above)
Comment (AndrejNikolicEq) [reply storybook/src/stories/person/card-example.stories.tsx:91]: nope, not working like that any more, can't resolve image without azureId (for now)

File: storybook/src/stories/person/list-item-component.stories.mdx (hunk 4 above)
Comment (AndrejNikolicEq) [reply storybook/src/stories/person/list-item-component.stories.mdx:73]: yes

File: packages/person/package.json (hunk 1 above)
Comment (AndrejNikolicEq) [reply packages/person/package.json:47]: can't, I have pretty on doc save

File: storybook/src/stories/person/resolve-mock/PersonResolve.tsx (hunk 5 above)
Comment (AndrejNikolicEq) [reply storybook/src/stories/person/resolve-mock/PersonResolve.tsx:177]:
```ts
"editor.formatOnSave": true,
"eslint.format.enable": true,
```

File: storybook/src/stories/person/select-example.stories.tsx (hunk 6 above)
Comment (AndrejNikolicEq) [reply storybook/src/stories/person/select-example.stories.tsx:87]:
```ts
"editor.formatOnSave": true,
"eslint.format.enable": true,
```

**Comment by odinr** [comment]
@equinor/fusion-wc-person@1.2.2 was just release, could you include that

File: packages/person/src/PersonProvider.tsx [hunk 7]
```diff
@@ -4,7 +4,7 @@ import { PersonProviderElement, PersonResolver } from '@equinor/fusion-wc-person
 /** Person porvider properties for the resolver */
 export type PersonProviderProps = {
   /** Person resolver interface that contains getPresence and getDetails functions */
-  resolve: PersonResolver;
+  personResolver: PersonResolver;
```
Comment (odinr) [inline packages/person/src/PersonProvider.tsx:7]: why is this renamed

File: packages/person/src/PersonSelect.tsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import { PropsWithChildren, useEffect, useRef } from 'react';
+import { ComponentProps, createComponent } from '@equinor/fusion-react-utils';
+import HTMLPersonSelectCustomElement, { tag } from '@equinor/fusion-wc-person/select';
+import extractProps from './extract-props';
+
+export type ElementProps = PropsWithChildren<
+  Partial<Pick<HTMLPersonSelectCustomElement, "autofocus" | "disabled" | "dropdownHeight" | "graphic" | "initialText" | "label" | "leadingIcon" | "meta" | "multiple" | "placeholder" | "selectedId" | "value" | "variant">>
```
Comment (odinr) [inline packages/person/src/PersonSelect.tsx:7]: fix linting

File: packages/person/src/index.tsx [hunk 9]
```diff
@@ -2,6 +2,7 @@ export * from './PersonAvatar';
 export * from './PersonCard';
 export * from './PersonProvider';
 export * from './PersonListItem';
+export * from './PersonSelect';
 export type { PersonResolver } from '@equinor/fusion-wc-person/provider';
 export { PersonAvailability, PersonAccountType } from '@equinor/fusion-wc-person';
-export type { PersonDetails, PersonPresence, PersonPicture } from '@equinor/fusion-wc-person';
+export type { PersonDetails, PersonPresence, PersonPicture, PersonInfo, PersonSearchResult } from '@equinor/fusion-wc-person';
```
Comment (odinr) [inline packages/person/src/index.tsx:8]: fix lint

File: packages/person/src/index.tsx [hunk 10]
```diff
@@ -2,6 +2,7 @@ export * from './PersonAvatar';
 export * from './PersonCard';
 export * from './PersonProvider';
 export * from './PersonListItem';
+export * from './PersonSelect';
```
Comment (odinr) [inline packages/person/src/index.tsx:5]: please use named exports

File: storybook/src/stories/person/avatar-component.stories.mdx [hunk 11]
```diff
@@ -60,18 +60,35 @@ You can create your own [`<PersonProvider>`](?path=/docs/data-person-provider--p
 
 Name                      | Type                            | Description
 ------------------------- | -----------------------------   | -----------
-`resolve`                 | `PersonResolver`               | Person resolver for a provider.
+`personResolver`          | `PersonResolver`                | Person resolver for a provider.
```
Comment (odinr) [inline storybook/src/stories/person/avatar-component.stories.mdx:63]: why renamed?

File: storybook/src/stories/person/avatar-component.stories.mdx [hunk 12]
```diff
@@ -60,18 +60,35 @@ You can create your own [`<PersonProvider>`](?path=/docs/data-person-provider--p
...
 
 ### Methods for PersonAvatar
```
Comment (odinr) [inline storybook/src/stories/person/avatar-component.stories.mdx:65]: this are methods for the provider, not the element

**Review by odinr — CHANGES_REQUESTED** [review]
azureids need to be guid
provider should only be named provider
props should make sense

File: storybook/src/stories/person/avatar-component.stories.mdx (hunk 12 above)
Comment (AndrejNikolicEq) [reply storybook/src/stories/person/avatar-component.stories.mdx:65]: Yes, they are in Person Provider section, but what I mean with title is these are methods used in provider for person card

**Review by odinr — APPROVED** [review]
